PrimeNumbers.java

```diff
@@ -16,18 +16,60 @@ public static void myassert(boolean  x) {
 
     private static int secondMaxSixDigitPrime(){
         //WRITE YOUR CODE HERE
-    }
+    	int count=0;
+    	int numFound =0;
+    	
+    	for(int n=999999 ; n >=100001 ; n-=2){
+    		boolean flag = false ;
+     		for(int m =2; m <= n / 2 ; m++){
+     			if(n % m ==0   ){
+    				flag =true;
+       				break;    			
+    			}
+        		    	
+    		}
+     		
+     		if (!flag){
+     		
+     			count++;
+     			if(count==2){
+     					numFound=n;
+     					break;
+     			}
+     			//System.out.println(n);
+     		}
+    	}
+    	return  numFound;
+}
 
 
     private static List<Integer> fourDigitPrimes(){
-        //WRITE YOUR CODE HERE
+    	
+          //WRITE YOUR CODE HERE
+    	ArrayList<Integer> primeList = new ArrayList<Integer>();
+    	
+    	for(int n=1000 ; n <=9999 ; n+=1){
+    		boolean primeFound = false ;
+     		for(int m =2; m <= n / 2 ; m++){
```

Comment:
Can we reuse this code, the prime # logic, since same logic or code is used in both the functions. Code reusability and also putting into it function always helps!